### ISSUE TITLE
Refactor the tearDownListeners() behavior for the command classes (aka. sometime "'batt' : commande inconnue. Envoyer "?" afficher l'aide. " occurs )

### ISSUE THREAD

**Issue by GoogleCodeExporter** [baluragala/gtalksms#189]
```
GTsms
'batt' : commande inconnue. Envoyer "?" afficher l'aide.
'batt' : commande inconnue. Envoyer "?" afficher l'aide.
Nouveau message de disponibilité de sms - GTalkSMS - 54% - Battery   23:26
Nouveau message de disponibilité de sms - GTalkSMS - 52% - Battery   23:26
```

Original issue reported on code.google.com by `lpenou.tstXX@mpo.li` on 17 Jun 2011 at 9:29

**Comment by GoogleCodeExporter** [comment]
```
I've seen this error on another person logs, i'll check if it's happening when 
the connection is lost
```

Original comment by `Florent....@gmail.com` on 17 Jun 2011 at 9:35
- Changed state: **Investigating**

**Comment by GoogleCodeExporter** [comment]
```
After reviewing the code, this defect could happen if right after a 
reconnection we receive a presence message but the command set has not been 
rebuilt yet.

XmppBuddies::presenceChanged send command "batt"
But command are rebuilt on each xmpp connected status in 
MainService::updateListenersToCurrentState

Todo : better command management on disconnection 
```

Original comment by `Florent....@gmail.com` on 17 Jun 2011 at 10:17

**Comment by GoogleCodeExporter** [comment]
```
As of now GTalkSMS does save messages when in a state of not CONNECTED or 
DISCONNECTED, I think we no longer need the old behavior of disabling the 
listeners by destroying all command instances and instead manage it with an an 
Observer DP or so.

```

Original comment by `fschm...@gmail.com` on 18 Jun 2011 at 8:57
- Changed title: **Refactor the tearDownListeners() behavior for the command classes (aka. sometime "'batt' : commande inconnue. Envoyer "?" afficher l'aide. " occurs )**
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision 57394427f6bd.
```

Original comment by `fschm...@gmail.com` on 20 Jun 2011 at 2:06
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
Ahh, so google code tries to be smart here. But I'd like this status more.
```

Original comment by `fschm...@gmail.com` on 20 Jun 2011 at 2:12
- Changed state: **WaitingForRelease**

**Comment by GoogleCodeExporter** [comment]
```
Fixed in v3.0
```

Original comment by `Florent....@gmail.com` on 15 Jul 2011 at 3:35
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
Original comment by `fschm...@gmail.com` on 25 Aug 2011 at 7:50
- Changed state: **Deployed**
